tsd/s2_metadata_parser.py

```diff
@@ -329,10 +330,14 @@ def build_gs_links(self):
         if 'granule_date' not in self:
             tile_info = get_roda_metadata(self, filename='tileInfo.json')
             #self.granule_date = dateutil.parser.parse(tile_info['timestamp'])
+            if not tile_info:  # abort if file not found on roda
+                return
             self.granule_date = parse_datastrip_id_for_granule_date(tile_info['datastrip']['id'])
 
         if 'absolute_orbit' not in self:
             product_info = get_roda_metadata(self, filename='productInfo.json')
+            if not tile_info:  # abort if file not found on roda
```

Comment:
I think there is a typo here: do you mean `product_info` instead of `tile_info`?

Reply:
oops! yes of course. Thanks!